### PR TITLE
Fix empty password change

### PR DESCRIPTION
**Pull request by TonyGiorgio**

### CODE REVIEW

File: mutiny-wasm/src/lib.rs [hunk 1]
```diff
@@ -1170,9 +1170,9 @@ impl MutinyWallet {
         old_password: Option<String>,
         new_password: Option<String>,
     ) -> Result<(), MutinyJsError> {
-        self.inner
-            .change_password(old_password, new_password)
-            .await?;
+        let old_p = old_password.filter(|p| !p.is_empty());
+        let new_p = new_password.filter(|p| !p.is_empty());
+        self.inner.change_password(old_p, new_p).await?;
```
Comment (benthecarman) [inline mutiny-wasm/src/lib.rs:1175]: we should probably be doing this deeper too

Comment (TonyGiorgio) [reply mutiny-wasm/src/lib.rs:1175]: We should really just remove `Option<String>` to be honest. We do these checks in so many places....

Comment (TonyGiorgio) [reply mutiny-wasm/src/lib.rs:1175]: But no, I would rather all the incorrect string parsing logic only happen in the wasm layer.

Comment (benthecarman) [reply mutiny-wasm/src/lib.rs:1175]: idk it is a better abstraction and works fine, it is just weird for js

Comment (benthecarman) [reply mutiny-wasm/src/lib.rs:1175]: looked through it, seems like this is the best spot actually

**Review by benthecarman — APPROVED** [review]

**Pull request merged**